src/NHibernate/Async/Engine/ForeignKeys.cs

```diff
@@ -171,38 +167,25 @@ public static async Task<bool> IsNotTransientSlowAsync(string entityName, object
 		/// <remarks>
 		/// Don't hit the database to make the determination, instead return null; 
 		/// </remarks>
-		public static Task<bool?> IsTransientFastAsync(string entityName, object entity, ISessionImplementor session, CancellationToken cancellationToken)
+		public static async Task<bool?> IsTransientFastAsync(string entityName, object entity, ISessionImplementor session, CancellationToken cancellationToken)
 		{
-			if (cancellationToken.IsCancellationRequested)
+			cancellationToken.ThrowIfCancellationRequested();
+			if (Equals(Intercept.LazyPropertyInitializer.UnfetchedProperty, entity))
 			{
-				return Task.FromCanceled<bool?>(cancellationToken);
+				// an unfetched association can only point to
+				// an entity that already exists in the db
+				return false;
 			}
-			try
-			{
-				if (Equals(Intercept.LazyPropertyInitializer.UnfetchedProperty, entity))
-				{
-					// an unfetched association can only point to
-					// an entity that already exists in the db
-					return Task.FromResult<bool?>(false);
-				}
 
-				// let the interceptor inspect the instance to decide
-				var isTransient = session.Interceptor.IsTransient(entity);
-				if (isTransient.HasValue)
-					return Task.FromResult<bool?>(isTransient);
+			if (entity.IsProxy())
+				return false;
 
-				if (entity is INHibernateProxy proxy && proxy.HibernateLazyInitializer.IsUninitialized)
-				{
-					return Task.FromResult<bool?>(false);
-				}
+			if (session.PersistenceContext.IsEntryFor(entity))
```

Comment:
Actually as shows NH607 this check is wrong. PersistenceContext can contain entries for transient objects.
And hack was added to support wrong implementation of `IsNotTransientSlow`:
https://github.com/nhibernate/nhibernate-core/blob/ac39173567b31bcfad475ca32687c9faf0d37f87/src/NHibernate/Engine/ForeignKeys.cs#L249-L255

So in this PR I'm going to move this check back to `IsNotTransientSlow` and in new PR for 5.3 will try to remove it completly.